Makefile

```diff
@@ -0,0 +1,91 @@
+.PHONY: default
+default: all
+
+DRIVER_NAME = CaptainJack.driver
+
+CC         := $(CC)
+CFLAGS      = -std=c99 -g3 -Wall -Wextra -Werror -Wno-unused-parameter -mmacosx-version-min=10.9
+CPPFLAGS    =
+LDFLAGS     =
+
+CFLAGS_CJ  = -bundle -framework CoreAudio -framework CoreFoundation
+CFLAGS_CJD = -framework CoreFoundation
+DESTDIR     =
+PREFIX      = /usr/local
+PLUGINDIR  = /Library/Audio/Plug-Ins/HAL/
+BUILDDIR    = build
+
+SRCS        = $(wildcard **/*.c)
+DEPS        = $(patsubst src/%,$(BUILDDIR)/%,$(addsuffix .d,$(SRCS)))
+
+-include $(BUILDDIR)/Makefile.dep
+
+# Dependencies
+
+$(BUILDDIR)/Makefile.dep: $(DEPS)
+	@mkdir -p $(dir $(@))
+	@cat $^ > $@
+
+$(BUILDDIR)/%.d: src/%
+	@mkdir -p $(dir $(@))
+	@$(CC) $(CPPFLAGS) -MG -MM $^ -MF $@ -MQ $(subst .c.d,.o,$@)
+
+$(BUILDDIR)/%.o: src/%.c
+	@mkdir -p $(dir $(@))
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
+
+# Targets
+
+$(BUILDDIR)/captain-jack-daemon: $(BUILDDIR)/captain-jack-daemon.o $(BUILDDIR)/xmit.o
+	$(CC) $(LDFLAGS) $(CFLAGS_CJD) $^ -o $@
+
+$(BUILDDIR)/captain-jack: $(BUILDDIR)/captain-jack-device.o $(BUILDDIR)/xmit.o
+	$(CC) $(LDFLAGS) $(CFLAGS_CJ) $^ -o $@
+
+.PHONY: all
+all: $(BUILDDIR)/captain-jack-daemon $(BUILDDIR)/captain-jack
+
+.PHONY: clean
+clean:
+	rm -rf $(BUILDDIR)
+
+# Install
+.PHONY: stage
+stage: $(BUILDDIR)/captain-jack
+	install -m 0755 $(BUILDDIR)/captain-jack $(DRIVER_NAME)/Contents/MacOS/
+
+.PHONY: install
+install: $(BUILDDIR)/captain-jack-daemon stage
+	sudo install -m 0755 -o root -g wheel $(BUILDDIR)/captain-jack-daemon $(DESTDIR)$(PREFIX)/sbin
+	sudo install -m 0644 -o root -g wheel $(DRIVER_NAME)/Contents/Info.plist $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Info.plist
+	sudo install -m 0755 -o root -g wheel $(DRIVER_NAME)/Contents/MacOS/CaptainJack $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/MacOS/CaptainJack
+	sudo install -m 0644 -o root -g wheel $(DRIVER_NAME)/Contents/Resources/DeviceIcon.icns $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Resources/DeviceIcon.icns
+	sudo install -m 0644 -o root -g wheel $(DRIVER_NAME)/Contents/Resources/English.lproj/Localizable.strings $(DESTDIR)$(PLUGINDIR)/$(DRIVER_NAME)/Contents/Resources/English.lproj/Localizable.strings
+	sudo install -m 0600 -o root -g wheel me.junon.CaptainJack.plist $(DESTDIR)/Library/LaunchDaemons/
```

Comment:
If the destination directory doesn't exist, this fails. Apparently `install -D` on other platforms works but OS/X doesn't seem to have that flag. Looks like it'll have to be a `mkdir -p` call.